js/ui/grid_core/ui.grid_core.column_chooser.js

```diff
@@ -369,38 +370,38 @@ const columnChooserMembers = {
         this._columnChooserList.option('items', items);
     },
 
-    _updateItemSelection: function(columnIndex) {
-        const isRecursive = this.option('columnChooser.selection.recursive');
-        const column = this._columnsController.columnOption(columnIndex);
+    _updateItemsSelection: function(columnIndices) {
+        const columnChooserList = this._columnChooserList;
+        const changedColumns = columnIndices?.map((columnIndex) => this._columnsController.columnOption(columnIndex));
 
-        const isRecursiveWithColumns = isRecursive && column.hasColumns;
-
-        if(!isRecursiveWithColumns) {
-            column.visible
-                ? this._columnChooserList.selectItem(columnIndex)
-                : this._columnChooserList.unselectItem(columnIndex);
-        }
+        columnChooserList.beginUpdate();
+        changedColumns?.forEach((changedColumn) => {
+            if(changedColumn.visible) {
+                columnChooserList.selectItem(changedColumn.index);
+            } else {
+                columnChooserList.unselectItem(changedColumn.index);
+            }
+        });
```

Comment:
Minor comment.

Let's use destructuring here:
```suggestion
        changedColumns?.forEach(({ visible, index }) => {
            if(visible) {
                this._columnChooserList.selectItem(index);
            } else {
                this._columnChooserList.unselectItem(index);
            }
        });
```